doc/source/reference/array_api.rst

```diff
@@ -0,0 +1,795 @@
+     - **Compatible**
+     - Equivalent to ``np.linalg.svd(compute_uv=False)``.
+   * - The ``axis`` keyword to ``tensordot`` must be a tuple.
+     - **Compatible**
+     - In ``np.tensordot``, it can also be an array or array-like.
+   * - ``trace`` operates on the last two axes.
+     - **Breaking**
+     - ``np.trace`` operates on the first two axes by default. Note that the
+       array API ``trace`` does not allow specifying which axes to operate on.
+
+Manipulation Functions Differences
+----------------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - Various functions have been renamed
+     - **Compatible**
+     - See :ref:`array_api-name-changes`.
+   * - ``concat`` has different default casting rules from ``np.concatenate``
+     - **Strictness**
+     - No cross-kind casting. No value-based casting on scalars.
+   * - ``stack`` has different default casting rules from ``np.stack``
+     - **Strictness**
+     - No cross-kind casting. No value-based casting on scalars.
+   * - New function ``permute_dims``.
+     - **Compatible**
+     - Unlike ``np.transpose``, the ``axis`` keyword argument to
+       ``permute_dims`` is required.
+   * - ``reshape`` function has a ``copy`` keyword argument
+     - **Compatible**
+     - See https://github.com/numpy/numpy/issues/9818.
+
+Set Functions Differences
+-------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - New functions ``unique_all``, ``unique_counts``, ``unique_inverse``,
+       and ``unique_values``.
+     - **Compatible**
+     - See :ref:`array_api-name-changes`.
+   * - The four ``unique_*`` functions return a named tuple.
+     - **Compatible**
+     -
+   * - ``unique_all`` and ``unique_indices`` return indices with the same
+       shape as ``x``.
+     - **Compatible**
+     - See https://github.com/numpy/numpy/issues/20638.
+
+.. _array_api-set-functions-differences:
+
+Set Functions Differences
+-------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - ``argsort`` and ``sort`` have a ``stable`` keyword argument instead of
+       ``kind``.
+     - **Compatible**
+     - ``stable`` is a boolean keyword argument, defaulting to ``True``.
+       ``kind`` takes a string, defaulting to ``"quicksort"``. ``stable=True``
+       is equivalent to ``kind="stable"`` and ``kind=False`` is equivalent to
+       ``kind="quicksort"`` (although any sorting algorithm is allowed by the
+       spec when ``stable=False``).
+   * - ``argsort`` and ``sort`` have a ``descending`` keyword argument.
+     - **Compatible**
+     -
+
+Statistical Functions Differences
+---------------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - ``sum`` and ``prod`` always upcast ``float32`` to ``float64`` when
+       ``dtype=None``.
+     - **Breaking**
```

Comment:
Yes, there's [no restriction on axis](https://data-apis.org/array-api/latest/API_specification/generated/signatures.statistical_functions.sum.html). It's not clear to me that the axis should matter? It wasn't ever mentioned in [the discussion](https://github.com/data-apis/array-api/issues/202) as far as I can tell.

Reply:
I faintly remember seeing the discussion and being surprised by it before, but not diving into it and thinking seriously about it.  I feel this change may be something to run by a wider audience.  I guess it may not be massive from a back-compat point of view if it only affects float16, float32, and complex64.

Also: I think this is at least the default float/integer precision when the input is float/integer?  Or do integer inputs return floats?

Reply:
The default output dtype is based on the input dtype: [spec](https://github.com/data-apis/array-api/blob/c58051748090fe97aae980837c63c02c5e405ff2/spec/API_specification/signatures/statistical_functions.py#L111).

If the input is an integer dtype, the default output dtype is the default integer dtype provided the default integer dtype has sufficient precision.

Similarly, if the input is a float dtype, the default output dtype is the default floating-point dtype, provided the default floating-point dtype has sufficient precision.

In short, when `dtype=None`, the output array should never have a dtype with less precision than the input dtype.